src/Animate/Animate.js

```diff
@@ -0,0 +1,131 @@
+  }
+
+  componentDidMount() {
+    const { triggerAnimation } = this.props;
+    return triggerAnimation ? this._toggleAnimation() : null;
+  }
+
+  componentDidUpdate(prevProps) {
+    const { triggerAnimation, loop } = this.props;
+
+    if (
+      prevProps.triggerAnimation !== triggerAnimation ||
+      prevProps.loop !== loop
+    ) {
+      this._toggleAnimation();
+    }
+  }
+
+  _onAnimationStart = () => {
+    const { onEnter } = this.props;
+    this.setState(
+      {
+        animationFinished: false,
+      },
+      () => onEnter && onEnter(),
+    );
+  };
+
+  _onAnimationEnd = () => {
+    const { onExited } = this.props;
+    this.setState(
+      {
+        animationFinished: true,
+        classNames: '',
+      },
+      () => onExited && onExited(),
+    );
+  };
+
+  _toggleAnimation = () => {
+    const size = this._getChildComponentSize();
+    const { classNames } = this.state;
+    const { loop } = this.props;
+
+    this.setState({
+      classNames: classNames ? '' : { ...styles('animation', { loop, size }) },
+    });
+  };
+
+  _renderFirstChild = () => {
+    const { children } = this.props;
+    const childrenArray = React.Children.toArray(children);
+    return childrenArray[0] ? childrenArray[0] : null;
+  };
+
+  _getChildComponentSize = () => {
+    const childWidth = this.childObserverRef.current.offsetWidth;
+    const { from, to } = childWidthRange.medium;
+
+    return inRange(childWidth, from, to) ? childSize.medium : childSize.small;
+  };
+
+  render() {
+    const { classNames } = this.state;
+    const { dataHook, delay } = this.props;
+
+    return (
+      <div
+        data-hook={dataHook}
+        {...classNames}
+        style={delay ? { animationDelay: delay } : null}
+        onAnimationStart={this._onAnimationStart}
+        onAnimationEnd={this._onAnimationEnd}
+      >
+        <div
+          data-hook={dataHooks.animateContent}
+          className={styles.animateContent}
+        >
+          <div className={styles.observer} ref={this.childObserverRef} />
+          {this._renderFirstChild()}
+        </div>
+      </div>
+    );
+  }
+}
+
+Animate.displayName = 'Animate';
+
+Animate.propTypes = {
+  /** Applied as data-hook HTML attribute that can be used in the tests */
+  dataHook: PropTypes.string,
+
+  /** The component which we would like to animate. When given multiple components- only the first child node will be animated. */
+  children: PropTypes.node.isRequired,
+
+  /** Triggers the animation transition */
+  triggerAnimation: PropTypes.bool.isRequired,
+
+  /** A callback fired immediately after the transition enters.*/
+  onEnter: PropTypes.func,
+
+  /** A callback fired immediately after the transition finishes.*/
+  onExited: PropTypes.func,
+
+  /** when set to true, the child component animate repetitively until stopped by other event*/
+  loop: PropTypes.bool,
+
+  /** set a delay before the animation execution */
+  delay: PropTypes.string,
```

Comment:
this should support a number also imo. number can be treated as `ms`
```suggestion
  delay: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```